policy-controller/core/src/http_route.rs

```diff
@@ -0,0 +1,125 @@
+use ahash::AHashMap as HashMap;
+use anyhow::Result;
+pub use hyper::http::{uri::Scheme, Method, StatusCode};
+use regex::Regex;
+
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct HttpRoute {
+    pub hostnames: Vec<Hostname>,
+    pub rules: Vec<HttpRouteRule>,
+}
+
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub enum Hostname {
+    Exact(String),
+    Suffix { reverse_labels: Vec<String> },
+}
+
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct HttpRouteRule {
+    pub matches: Vec<HttpRouteMatch>,
+    pub filters: Vec<HttpFilter>,
+}
+
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub enum HttpFilter {
+    RequestHeaderModifier {
+        add: HashMap<String, String>,
+        set: HashMap<String, String>,
+        remove: Vec<String>,
```

Comment:
should these be represented as `http::HeaderName`/`HeaderValue` so that we can validate that they are valid headers? i notice we do this for most of the URI components in `RequestRedirect`, but not for headers...
```suggestion
        add: HashMap<HeaderName, HeaderValue>,
        set: HashMap<HeaderName, HeaderValue>,
        remove: Vec<HeaderName>,
```
(of course, we'd have to import `hyper::http::header::{HeaderName, HeaderValue}`)